### PR TITLE
feat(dedicated.server): improve server monitoring 

### PR DESCRIPTION
**Pull request by mohammed-zahaf**
| Question         | Answer
| ---------------- | ---
| Branch?          | `release/sprint-2238-2240`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-9918
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
1. add and manage new monitoring options:
1.1 disabled
1.2 proactive
1.3 no-proactive
2. implement tracking

## Translation

- [x] TRANS-46006

### CODE REVIEW

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/dashboard.html [hunk 1]
```diff
@@ -331,37 +331,22 @@
                             class="oui-badge"
                             data-ng-class="{
                                     'oui-badge_success': $ctrl.server.monitored,
-                                    'oui-badge_error': !$ctrl.server.monitored
+                                    'oui-badge_warning': !$ctrl.server.monitored
                                 }"
-                            data-translate="{{'server_configuration_monitoring_' + ($ctrl.server.monitored && 'activated' || 'disabled')}}"
-                            data-oui-tooltip="{{ ($ctrl.server.monitored && 'server_configuration_monitoring_activate_information' | translate) || ('server_configuration_monitoring_deactivate_information' | translate) }}"
+                            data-ng-bind="$ctrl.getMonitoringStatusTranslation()"
```
Comment (JeremyDec) [inline packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/dashboard.html:336]:
```suggestion
                            data-rtanslate="$ctrl.getMonitoringStatusTranslation()"
```
Method could return only the generated translation key

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/dashboard.controller.js [hunk 2]
```diff
@@ -245,6 +245,23 @@ export default class DedicatedServerDashboard {
       });
   }
 
+  getMonitoringStatus() {
+    const { monitored, noIntervention } = this.server;
+    let monitoringStatus = `disabled`;
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/dashboard.controller.js:250]:
```suggestion
    let monitoringStatus = 'disabled';
```

File: packages/manager/apps/dedicated/client/app/dedicated/server/monitoring/update/dedicated-server-monitoring-update.controller.js [hunk 3]
```diff
@@ -1,24 +1,53 @@
...
+          status: 'proactive',
+          settings: { monitoring: true, noIntervention: false },
+          tracking: 'enable-with-intervention',
+        },
+        {
+          status: 'no-proactive',
+          settings: { monitoring: true, noIntervention: true },
+          tracking: 'enable-without-intervention',
+        },
+      ];
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/server/monitoring/update/dedicated-server-monitoring-update.controller.js:44]: Can move in constants file ?

File: packages/manager/apps/dedicated/client/app/dedicated/server/monitoring/update/dedicated-server-monitoring-update.controller.js [hunk 4]
```diff
@@ -29,76 +58,102 @@ angular.module('App').controller(
         });
     }
 
+    getMonitoringStatus() {
+      const { monitored, noIntervention } = this.server;
+      let monitoringStatus = `disabled`;
```
Comment (ghyenne) [inline packages/manager/apps/dedicated/client/app/dedicated/server/monitoring/update/dedicated-server-monitoring-update.controller.js:63]:
```suggestion
      let monitoringStatus = 'disabled';
```

File: packages/manager/apps/dedicated/client/app/dedicated/server/monitoring/update/dedicated-server-monitoring-update.controller.js (hunk 3 above)
Comment (mohammed-zahaf) [reply packages/manager/apps/dedicated/client/app/dedicated/server/monitoring/update/dedicated-server-monitoring-update.controller.js:44]: No, it's a model data.

File: packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/dashboard.controller.js (hunk 2 above)
Comment (mohammed-zahaf) [reply packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/dashboard.controller.js]: Done

File: packages/manager/apps/dedicated/client/app/dedicated/server/monitoring/update/dedicated-server-monitoring-update.controller.js (hunk 4 above)
Comment (mohammed-zahaf) [reply packages/manager/apps/dedicated/client/app/dedicated/server/monitoring/update/dedicated-server-monitoring-update.controller.js]: Done

**Review by ghyenne — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/dedicated/server/monitoring/update/dedicated-server-monitoring-update.controller.js [hunk 5]
```diff
@@ -29,76 +58,109 @@ angular.module('App').controller(
...
+    }
+
     /**
      * Update monitoring.
      * @return {Promise}
      */
     update() {
-      const { monitored } = this.server;
+      this.trackClick('confirm');
+      this.trackConfirmClick(this.selectedOption.tracking);
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/dedicated/server/monitoring/update/dedicated-server-monitoring-update.controller.js:115]: **question:** Do we need 2 trackings for the same action?

Comment (mohammed-zahaf) [reply packages/manager/apps/dedicated/client/app/dedicated/server/monitoring/update/dedicated-server-monitoring-update.controller.js:115]: Yes required from tracking team. Already validated.

**Pull request merged**